app/models/ext_management_system.rb

```diff
@@ -917,6 +922,30 @@ def stop_event_monitor_queue_on_credential_change
     end
   end
 
+  def stop_workers_on_endpoint_change
+    return if new_record?
+    return if default_endpoint.changed.include_none?("hostname", "ipaddress")
+
+    _log.info("EMS: [#{name}], Hostname or IP address has changed, stopping workers.  They will be restarted by the WorkerMonitor.")
+
+    stop_workers_for_ems
+  end
+
+  def stop_workers_on_credential_change
+    return if new_record?
+    return unless credentials_changed?
```

Comment:
It would be nice to put in a few tests that changing the credentials do properly trigger the `credentials_changed?` I'm suspicious of `_changed?` for non string/integer fields